qiime2/core/validate.py

```diff
@@ -0,0 +1,192 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2021, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from qiime2.core.exceptions import ValidationError, ImplementationError
+from qiime2.core.transform import ModelType
+from qiime2.core.util import sorted_poset
+
+
+class ValidationObject:
+    r"""
+    Store, sort and run all semantic validators for a for a single, complete
+    semantic type(a `concrete type`).
+
+
+    Attributes
+    ----------
+    concrete_type: SemanticType
+        The semantic type for which the validators are valid for.
+
+    """
+    def __init__(self, concrete_type):
+        r"""
+        Parameters
+        ----------
+        concrete_type: semantic type
+            The single, complete semantic type that the validators are to be
+            associated with.
+
+        Attributes
+        ----------
+        _validators: list
+            A list of ValidatorRecords
+
+        _is_sorted: Bool
+            Tracks whether or not `_validators` has been sorted or not.
```

Comment:
Turn these attributes (`_validators` and `_is_sorted`) into a comment instead of public docs and this is good to go!